File: Platform/codesignal/Intro/32.absoluteValuesSumMinimization/solution.go
package main

import (
	"fmt"
	"math"
)

func solution(a []int) int {
	min, index := math.MaxInt64, 0
	for i := 0; i < len(a); i++ {
		sum := 0
		for j := 0; j < len(a); j++ {
			if i == j {
				continue
			}
			sum += int(math.Abs(float64(a[j] - a[i])))
		}
		if sum < min {
			min = sum
			index = i
		}
	}
	return a[index]
}

func main() {
	a := []int{2, 4, 7}
	fmt.Println(solution(a))
}

File: Platform/codesignal/Intro/32.absoluteValuesSumMinimization/solution.js
function solution(a) {
  let [min, index] =  [Number.MAX_VALUE, 0];
  for (let i=0; i<a.length; i++) {
    let sum = 0;
    for (let j=0; j<a.length; j++) {
      if (i === j) continue;
      sum += Math.abs(a[j] - a[i])
    }
    if (sum < min) {
      min = sum;
      index = i;
    }
  }
  return a[index];
}

function solution(A) {
  return A[Math.ceil(A.length/2)-1];
}


console.log(solution([2, 4, 7]));